shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -0,0 +1,647 @@
+          + "requestCode: " + requestCode + "\n"
+          + "resultCode: " + resultCode + "\n"
+          + "data: " + data);
+      flutterEngine.getActivityControlSurface().onActivityResult(requestCode, resultCode, data);
+    } else {
+      Log.w(TAG, "onActivityResult() invoked before FlutterFragment was attached to an Activity.");
+    }
+  }
+
+  /**
+   * Invoke this from {@link Activity#onUserLeaveHint()}.
+   * <p>
+   * A {@code Fragment} host must have its containing {@code Activity} forward this call so that
+   * the {@code Fragment} can then invoke this method.
+   * <p>
+   * This method forwards to interested Flutter plugins.
+   */
+  void onUserLeaveHint() {
+    ensureAlive();
+    if (flutterEngine != null) {
+      Log.v(TAG, "Forwarding onUserLeaveHint() to FlutterEngine.");
+      flutterEngine.getActivityControlSurface().onUserLeaveHint();
+    } else {
+      Log.w(TAG, "onUserLeaveHint() invoked before FlutterFragment was attached to an Activity.");
+    }
+  }
+
+  /**
+   * Invoke this from {@link Activity#onTrimMemory(int)}.
+   * <p>
+   * A {@code Fragment} host must have its containing {@code Activity} forward this call so that
+   * the {@code Fragment} can then invoke this method.
+   * <p>
+   * This method sends a "memory pressure warning" message to Flutter over the "system channel".
+   */
+  void onTrimMemory(int level) {
+    ensureAlive();
+    if (flutterEngine != null) {
+      // Use a trim level delivered while the application is running so the
+      // framework has a chance to react to the notification.
+      if (level == TRIM_MEMORY_RUNNING_LOW) {
+        Log.v(TAG, "Forwarding onTrimMemory() to FlutterEngine. Level: " + level);
+        flutterEngine.getSystemChannel().sendMemoryPressureWarning();
+      }
+    } else {
+      Log.w(TAG, "onTrimMemory() invoked before FlutterFragment was attached to an Activity.");
+    }
+  }
+
+  /**
+   * Invoke this from {@link Activity#onLowMemory()}.
+   * <p>
+   * A {@code Fragment} host must have its containing {@code Activity} forward this call so that
+   * the {@code Fragment} can then invoke this method.
+   * <p>
+   * This method sends a "memory pressure warning" message to Flutter over the "system channel".
+   */
+  void onLowMemory() {
+    Log.v(TAG, "Forwarding onLowMemory() to FlutterEngine.");
+    ensureAlive();
+    flutterEngine.getSystemChannel().sendMemoryPressureWarning();
+  }
+
+  /**
+   * Ensures that this delegate has not been {@link #release()}'ed.
+   * <p>
+   * An {@code IllegalStateException} is thrown if this delegate has been {@link #release()}'ed.
+   */
+  private void ensureAlive() {
+    if (host == null) {
+      throw new IllegalStateException("Cannot execute method on a destroyed FlutterActivityAndFragmentDelegate.");
+    }
+  }
+
+  /**
+   * The {@link FlutterActivity} or {@link FlutterFragment} that owns this
+   * {@code FlutterActivityAndFragmentDelegate}.
+   */
+  interface Host {
```

Comment:
nit: add /* package */ or internal/default/wtv so people know it's intentional and don't accidentally change it later

Reply:
done